python_modules/dagster/dagster/_core/definitions/result.py

```diff
@@ -1,30 +1,35 @@
 from collections.abc import Mapping, Sequence
-from typing import NamedTuple, Optional
+from datetime import datetime  # noqa
+from os import PathLike  # noqa
+from typing import Generic, Optional, TypeVar
+
+from dagster_shared.check.decorator import checked
 
 import dagster._check as check
 from dagster._annotations import PublicAttr
 from dagster._core.definitions.asset_check_result import AssetCheckResult
 from dagster._core.definitions.data_version import DataVersion
 from dagster._core.definitions.events import AssetKey, CoercibleToAssetKey
-from dagster._core.definitions.metadata import RawMetadataMapping
-
-
-class AssetResult(
-    NamedTuple(
-        "_AssetResult",
-        [
-            ("asset_key", PublicAttr[Optional[AssetKey]]),
-            ("metadata", PublicAttr[Optional[RawMetadataMapping]]),
-            ("check_results", PublicAttr[Sequence[AssetCheckResult]]),
-            ("data_version", PublicAttr[Optional[DataVersion]]),
-            ("tags", PublicAttr[Optional[Mapping[str, str]]]),
-        ],
-    )
-):
+from dagster._core.definitions.metadata import (  # noqa
```

Comment:
@alangenfeld this is on account of me using `@checked` for the class constructors, and the constituent parts of the `RawMetadataMapping` union not being available in local scope, causing me to need to import "unused" symbols. Curious if you have thoughts on a better way to handle this (also I don't need to use `@checked`, just thought it might be nice).